src/mbgl/map/map_context.cpp

```diff
@@ -312,6 +312,8 @@ void MapContext::renderStill(const TransformState& state, const FrameData& frame
 MapContext::RenderResult MapContext::renderSync(const TransformState& state, const FrameData& frame) {
     assert(util::ThreadContext::currentlyOn(util::ThreadType::Map));
 
+    view.beforeRender();
```

Comment:
This should go after the early exit condition `if (!style)` so that `beforeRender` is guaranteed to be paired with a corresponding `afterRender`.